### PR TITLE
Setup release process

### LINKED ISSUE #74: Set up automated release process

### PR DESCRIPTION
**Pull request by sigiesec**
Fixes #74

### CODE REVIEW

**Comment by sigiesec** [comment]
@jcoeltjen Please give this a thorough look. It seems to work now, but maybe some things are done in an unusual way or are unnecessary.

File: deploy.sh [hunk 1]
```diff
@@ -1,25 +1,79 @@
...
+        if [ "$TRAVIS_BRANCH" != "master" ] ; then
+            BRANCH_FRAGMENT="-${TRAVIS_BRANCH////-}"
+        else
+            BRANCH_FRAGMENT=""
+        fi
+    else
+        BRANCH_FRAGMENT="-pr${TRAVIS_PULL_REQUEST}"
+    fi
+
+    cd com.btc.serviceidl.plainjava && mvn versions:set -DnewVersion=${BASE_VERSION}${BRANCH_FRAGMENT}${SNAPSHOT_FRAGMENT}
```
Comment (jcoeltjen) [inline deploy.sh]: Why do you build the new version from multiple fragments?
Is is not possible to use `RELEASE_VERSION` or `DEVELOPMENT_VERSION` for this?

File: deploy.sh [hunk 2]
```diff
@@ -1,25 +1,79 @@
...
 fi
 
-if [ "$TRAVIS_PULL_REQUEST" == "false" ] ; then
-    if [ "$TRAVIS_BRANCH" != "master" ] ; then
-        BRANCH_FRAGMENT="-${TRAVIS_BRANCH////-}"
-    else
-        BRANCH_FRAGMENT=""
+cp .travis.settings.xml $HOME/.m2/settings.xml
+
+if [[ "$TRAVIS_BRANCH" =~ ^release/ ]] ; then
```
Comment (jcoeltjen) [inline deploy.sh]: Is `TRAVIS_BRANCH` set if building a PR?
If not you can skip the check in line 55

File: deploy.sh [hunk 3]
```diff
@@ -1,25 +1,79 @@
...
 
-mvn deploy -DskipTests
+    # Checkout branch (travis-ci has checked out the commit hash)
+    # Note: if the branch has been updated in between, this will use a newer 
+    # revision than before, but for making the release this should not usually 
+    # happen, and should not matter anyway, since the newer revision would be 
+    # used by a subsequent build and overwrite the tag
+    git checkout $TRAVIS_BRANCH
+
+    DEVELOPMENT_VERSION=$(python -c 'version = tuple(map(int, ("'${RELEASE_VERSION}'".split(".")))); print("%i.%i.%i" % (version[0], version[1]+1, 0))')
```
[the remark this answers is not in the document]
Comment (sigiesec) [reply deploy.sh]: My sed knowledge is quite limited. I copied the sed expression from somewhere, but I wrote this myself ;) In addition, I think incrementing a number is not straight-forward with sed.

File: deploy.sh [hunk 4]
```diff
@@ -1,25 +1,79 @@
...
+        else
+            BRANCH_FRAGMENT=""
+        fi
+    else
+        BRANCH_FRAGMENT="-pr${TRAVIS_PULL_REQUEST}"
+    fi
+
+    cd com.btc.serviceidl.plainjava && mvn versions:set -DnewVersion=${BASE_VERSION}${BRANCH_FRAGMENT}${SNAPSHOT_FRAGMENT}
+
+    mvn deploy -DskipTests
```
[the remark this answers is not in the document]
Comment (sigiesec) [reply deploy.sh]: Yes, if release:perform is not used, this could be done.

[the remark this answers is not in the document]
Comment (sigiesec) [reply deploy.sh]: I will leave it at is it for now. Maybe #181 will change this again.

**Pull request merged**